packages/interactivity/src/vdom.ts

```diff
@@ -118,11 +118,25 @@ export function toVdom( root ) {
 		if ( directives.length ) {
 			props.__directives = directives.reduce(
 				( obj, [ name, ns, value ] ) => {
-					const [ , prefix, suffix = 'default' ] =
-						directiveParser.exec( name );
-					if ( ! obj[ prefix ] ) {
-						obj[ prefix ] = [];
+					const directiveMatch = directiveParser.exec( name );
+					const prefix = directiveMatch ? directiveMatch[ 1 ] : '';
+					const suffix =
+						directiveMatch && directiveMatch[ 2 ]
+							? directiveMatch[ 2 ]
+							: 'default';
+
+					if (
+						! directiveMatch &&
+						// @ts-expect-error This is a debug-only warning.
+						typeof SCRIPT_DEBUG !== 'undefined' &&
+						// @ts-expect-error
+						SCRIPT_DEBUG === true
+					) {
+						// eslint-disable-next-line no-console
+						console.warn( `Invalid directive: ${ name }.` );
 					}
+
+					obj[ prefix ] = obj[ prefix ] || [];
```

Comment:
Still not sure if we want `obj['']` @DAreRodz

Reply:
I think that if the `directiveParser` fails, the attribute should not be considered a directive and should be skipped. Having an empty `prefix` means we don't know which directive the attribute is.

Reply:
Basically, what @sirreal has mentioned in https://github.com/WordPress/gutenberg/pull/61249#pullrequestreview-2034999124. 😄